homeassistant/components/mqtt/mixins.py

```diff
@@ -205,7 +210,7 @@ def validate_device_has_at_least_one_identifier(value: ConfigType) -> ConfigType
             ),
             vol.Optional(CONF_MANUFACTURER): cv.string,
             vol.Optional(CONF_MODEL): cv.string,
-            vol.Optional(CONF_NAME): cv.string,
+            vol.Optional(CONF_NAME): vol.Any(cv.string, None),
```

Comment:
Why do we allow setting the device name to `None`?

Reply:
I think that is an error